### DISCUSSION TITLE
install.json and manifest.json files location

### DISCUSSION THREAD

**Discussion by amreus** [ScoopInstaller/Scoop#4577] [Ideas]
These 2 files are being copied/created in the same directory where the app is extracted, which I will call the "app directory."  Technically the files belong to `scoop` and not to the app.  A such, they do not belong in the app directory. Would it make sense to create these files in a directory that scoop controls, such as one directory higher than the app directory?

Is there any potential that these copied files will over-write an existing file that is part of the app extraction?    

It looks like there are at least 4 files that may get copied or created when an app is installed:

* manifest.json
* install.json
* install-context.reg
* uninstall-context.reg

**Comment by rashil2000** [comment]
> Is there any potential that these copied files will over-write an existing file that is part of the app extraction?

We haven't seen any such case so far, so it's safe to assume they were intelligently chosen :)

The .reg files will get copied only in special cases and they are done manually in the manifest, so they won't cause conflicts.

**Comment by ymeine** [comment]
> Is there any potential that these copied files will over-write an existing file that is part of the app extraction?

If scoop blindly forces the existence of such files in a target folder where it does not control the rest of the content, the answer is obviously: yes, there's potential (that's a yes/no question).

Now, as for the **probability** for it to happen, the reply

> We haven't seen any such case so far, so it's safe to assume they were intelligently chosen :)

seems to be empirically answering: "probability is extremely low".

But I wouldn't brag on suggesting they were "intelligently chosen"... When this feature was implemented, no one had long-term data allowing to say a such thing, so it was a pure bet.

And even with that assessment, the issue is still there: any new application has the potential to have conflicting files.

Is it a deliberate choice to keep that now known issue in the code? Or not to open a follow-up GitHub issue?

Especially since the fix seems quite trivial: dedicate an isolated folder, external to the application tree, to put all extra files. Proper data isolation at all levels is the ABC of software development.
